ICSharpCode.CodeConverter/VB/MethodBodyExecutableStatementVisitor.cs

```diff
@@ -294,6 +294,10 @@ public override SyntaxList<StatementSyntax> VisitForStatement(CSS.ForStatementSy
             }
             return SyntaxFactory.SingletonList(block);
         }
+        private static StatementSyntax ReplaceExitForToWhile(StatementSyntax statement) {
+            var exitSyntaxes = statement.DescendantNodesAndSelf().OfType<ExitStatementSyntax>();
+            return statement.ReplaceNodes(exitSyntaxes, (_, __) => SyntaxFactory.ExitWhileStatement());
```

Comment:
Using any of the Replace functions on the conversion result from another Visit method causes issues keeping track of where nodes came from. On PR #512 I'm currently unpicking existing ones of these to try to get comment conversion working C# -> VB.

I've added a commit to replace this with something that also works for nested loops and the continue statement. Feel free to build on that if you can think of other cases.